tests/integration/heartbeat/heartbeat.spec.ts

```diff
@@ -1,89 +1,80 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { DependencyContainer } from 'tsyringe';
+import { DataSource } from 'typeorm';
 import httpStatusCodes from 'http-status-codes';
-import { container } from 'tsyringe';
 import jsLogger from '@map-colonies/js-logger';
 import { trace } from '@opentelemetry/api';
-import { HeartbeatEntity } from '../../../src/DAL/entity/heartbeat';
-import { HeartbeatRepository } from '../../../src/DAL/repositories/heartbeatRepository';
-import { RepositoryMocks, initTypeOrmMocks, registerRepository, lessThanMock } from '../../mocks/DBMock';
+import { HeartbeatRepository, HEARTBEAT_REPOSITORY_SYMBOL } from '../../../src/DAL/repositories/heartbeatRepository';
 import { getApp } from '../../../src/app';
 import { SERVICES } from '../../../src/common/constants';
 import { HeartbeatRequestSender } from './helpers/requestSender';
 
-let heartbeatRepositoryMocks: RepositoryMocks;
-
-const now = 1620894317;
-const nowDate = new Date(now);
-
 describe('heartbeat', function () {
   let requestSender: HeartbeatRequestSender;
-  beforeAll(() => {
-    jest.useFakeTimers();
-    jest.setSystemTime(nowDate);
-  });
-
-  beforeEach(function () {
-    initTypeOrmMocks();
-    const app = getApp({
+  let repo: HeartbeatRepository;
+  let depContainer: DependencyContainer;
+  let saveSpy: jest.SpyInstance;
+  let findSpy: jest.SpyInstance;
+  beforeAll(async function () {
+    const [app, container] = await getApp({
       override: [
         { token: SERVICES.LOGGER, provider: { useValue: jsLogger({ enabled: false }) } },
         { token: SERVICES.TRACER, provider: { useValue: trace.getTracer('testTracer') } },
       ],
-      useChild: false, //child container is incompatible with the typeorm repositories implementation
+      useChild: true,
     });
-    heartbeatRepositoryMocks = registerRepository(HeartbeatRepository, new HeartbeatRepository());
     requestSender = new HeartbeatRequestSender(app);
+    depContainer = container;
+    repo = depContainer.resolve(HEARTBEAT_REPOSITORY_SYMBOL);
+  });
+
+  beforeEach(function () {
+    saveSpy = jest.spyOn(repo, 'save');
+    findSpy = jest.spyOn(repo, 'find');
   });
 
   afterEach(function () {
-    container.clearInstances();
+    jest.restoreAllMocks();
     jest.resetAllMocks();
   });
 
-  afterAll(() => {
-    jest.useRealTimers();
+  afterAll(async function () {
+    await depContainer.resolve(DataSource).destroy();
   });
 
   describe('Happy Path', function () {
+    it('pulse should return 200 status code and save the heartbeat pulse', async function () {
+      const id = '1';
+      saveSpy = jest.spyOn(repo, 'save');
+      const response = await requestSender.pulse(id);
+
+      //expect(response).toSatisfyApiSpec();
```

Comment:
Remove this comment

Reply:
revert remark keep code

Reply:
added a to do due to the bug we discussed